internal/clients/compliance/client.go

```diff
@@ -126,6 +131,48 @@ func (cc *ComplianceClient) PolicyDataForMinorVersion(ctx context.Context, major
 }
 
 func (cc *ComplianceClient) PolicyCustomizations(ctx context.Context, majorVersion, minorVersion int, policyID string) (*blueprint.Blueprint, error) {
+	// First check if the tailorings exist for the policy - this endpoint returns 200 OK with an empty array for `data` field if there are no tailorings
+	tailoringsGetResp, err := cc.request(ctx, "GET", fmt.Sprintf("%s/policies/%s/tailorings", cc.url, policyID))
+	if err != nil {
+		return nil, err
+	}
+	defer tailoringsGetResp.Body.Close()
+
+	var tailoringsResponse v2TailoringsResponse
+	err = json.NewDecoder(tailoringsGetResp.Body).Decode(&tailoringsResponse)
+	if err != nil {
+		return nil, err
+	}
+
+	if tailoringsGetResp.StatusCode == http.StatusOK && len(tailoringsResponse.Data) == 0 {
+		// If there are no tailorings for the policy, we need to create one, otherwise the request for tailoring file will return 404
+		// see https://issues.redhat.com/browse/RHINENG-16877
+		var jsonStr = []byte(fmt.Sprintf(`{"os_minor_version": %d}`, minorVersion))
+		tailoringsReq, err := http.NewRequest("POST", fmt.Sprintf("%s/policies/%s/tailorings", cc.url, policyID), bytes.NewBuffer(jsonStr))
```

Comment:
why not let this request go via the `request` helper? Should be easy enough to add POST support (accepting a `body io.Reader`, see the composer client for some inspiration).